src/passes/GlobalStructInference.cpp

```diff
@@ -417,29 +418,36 @@ struct GlobalStructInference : public Pass {
         // Helper for optimization: Given a Value, returns what we should read
         // for it.
         auto getReadValue = [&](const Value& value) -> Expression* {
+          Expression* ret;
           if (value.isConstant()) {
             // This is known to be a constant, so simply emit an expression for
             // that constant.
-            return value.getConstant().makeExpression(wasm);
-          }
+            ret = value.getConstant().makeExpression(wasm);
+          } else {
+            // Otherwise, this is non-constant, so we are in the situation where
+            // we want to un-nest the value out of the struct.new it is in. Note
+            // that for later work, as we cannot add a global in parallel.
 
-          // Otherwise, this is non-constant, so we are in the situation where
-          // we want to un-nest the value out of the struct.new it is in. Note
-          // that for later work, as we cannot add a global in parallel.
+            // There can only be one global in a value that is not constant,
+            // which is the global we want to read from.
+            assert(value.globals.size() == 1);
 
-          // There can only be one global in a value that is not constant, which
-          // is the global we want to read from.
-          assert(value.globals.size() == 1);
+            // Create a global.get with temporary name, leaving only the
+            // updating of the name to later work.
+            auto* get = builder.makeGlobalGet(value.globals[0],
```

Comment:
```suggestion
            ret = builder.makeGlobalGet(value.globals[0],
```
Nit: Can we do this and remove `ret = get` below?

Reply:
That almost works, but we need the type to be more specific than `Expression*` on line 441, where we use `get` in a place that must be `GlobalGet*`.